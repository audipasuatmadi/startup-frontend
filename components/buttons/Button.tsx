import React from 'react';
import BaseButton from './BaseButton';

type themes = 'brand' | 'add later';

export interface ButtonProps {
  disabled?: boolean;
  href?: string;
  children?: any;

  className?: string;
  type?: 'button' | 'submit' | 'reset';

  onClick?: (e: React.MouseEvent<HTMLButtonElement | HTMLAnchorElement>) => void;
  onMouseDown?: (e: React.MouseEvent<HTMLButtonElement | HTMLAnchorElement>) => void;

  rounded?: boolean;
  outlined?: boolean;
  theme?: themes;
}

/**
 * 
 * @version 0.1.0
 * @description basically wraps BaseButton, adding `outlined` and `rounded` features.
 * @param rounded boolean
 * @param outlined boolean
 * @author Putu Audi Pasuatmadi 
 */
const Button: React.FC<ButtonProps> = ({
  outlined,
  rounded,
  theme,
  disabled,
  className,
  ...otherProps
}: ButtonProps) => {
  const finalStyles = `button${rounded ? ' rounded-full' : ''}${
    disabled
      ? ` button-disabled bg-${theme}-dark text-white`
      : `${
          outlined
            ? ` bg-none border-2 border-${theme} text-${theme} hover:text-white`
            : ` bg-${theme} text-white`
        } hover:bg-${theme}-dark active:bg-${theme}-light focus:ring-4 ring-${theme} ring-opacity-50`
  } ${className}`;

  return (
    <BaseButton
      className={`${finalStyles}`}
      disabled={disabled}
      {...otherProps}
    />
  );
};

Button.defaultProps = {
  theme: 'brand'
}

export default Button;
