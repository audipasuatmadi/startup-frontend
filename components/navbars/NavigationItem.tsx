import React from 'react';
import Link from 'next/link';
import NavLink from './NavLink';

interface Props {
  href?: string;
  children: string | JSX.Element | null;
}

const NavigationItem = ({ children, href }: Props) => {
  if (children != null) {
    return (
      <li
        className='text-center'
        style={{
          height: 'max-content',
        }}>
        {typeof children === 'string' ? (
          <NavLink href={href ? href : '#'} activeClassName='font-semibold'>
            <a className='text-center text-gray-700 focus:outline-none bg-gray-500 bg-opacity-0 px-2 py-1 rounded focus:bg-opacity-50 transition-colors hover:bg-opacity-25'>
              {children}
            </a>
          </NavLink>
        ) : (
          children
        )}
      </li>
    );
  } else {
    return null;
  }
};

export default NavigationItem;
