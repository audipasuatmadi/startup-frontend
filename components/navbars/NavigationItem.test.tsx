import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import Button from '../buttons/Button';
import NavigationItem from './NavigationItem';


describe('navigationitem rendering tests', () => {
  it('should render other component correctly', () => {
    const wrapper: ShallowWrapper = shallow(
      <NavigationItem>
        <Button>hello</Button>
      </NavigationItem>
    );
    expect(wrapper.find('Button').exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
