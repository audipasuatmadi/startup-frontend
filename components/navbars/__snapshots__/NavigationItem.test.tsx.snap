// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigationitem rendering tests should render other component correctly 1`] = `
<li
  className="text-center"
  style={
    Object {
      "height": "max-content",
    }
  }
>
  <Button
    theme="brand"
  >
    hello
  </Button>
</li>
`;
